packages/office-ui-fabric-react/src/components/Overlay/Overlay.types.ts

```diff
@@ -1,19 +1,68 @@
 import * as React from 'react';
+import { OverlayBase } from './Overlay.base';
+import { IStyle, ITheme } from '../../Styling';
+import { IStyleFunction } from '../../Utilities';
 
 export interface IOverlay {
 
 }
 
 export interface IOverlayProps extends React.HTMLAttributes<HTMLElement> {
   /**
-   * Optional callback to access the IOverlay interface. Use this instead of ref for accessing
-   * the public methods and properties of the component.
+   * Gets the component ref.
    */
-  componentRef?: (component: IOverlay) => void;
+  componentRef?: (component: IOverlayProps) => void;
+
+  /**
+   * Call to provide customized styling that will layer on top of the variant rules
+   */
+  getStyles?: IStyleFunction<IOverlayStyleProps, IOverlayStyles>;
+
+  /**
+   * Theme provided by HOC.
+   */
+  theme?: ITheme;
+
+  /**
+   * Additional css class to apply to the Overlay
+   * @defaultvalue undefined
+   */
+  className?: string;
 
   /**
    * Whether to use the dark-themed overlay.
    * @defaultvalue false
    */
   isDarkThemed?: boolean;
+
+  onClick?: () => void;
+}
+
+export interface IOverlayStyleProps {
+  /**
+   * Accept theme prop.
+   */
+  theme: ITheme;
+
+  /**
+   * Accept custom classNames
+   */
+  className?: string;
+
+  /**
+   * Is overlay visible
+   */
+  isNone?: boolean;
+
+  /**
+   * Is overlay dark themed
+   */
+  isDarkThemed?: boolean;
+}
+
+export interface IOverlayStyles {
+  /**
+   * Style for the root element.
+   */
+  root?: IStyle;
```

Comment:
don't we want all classNames to be required? @dzearing @mikewheaton

Reply:
that way we can insure, inside getStyles, that we're making the entire thing. and then other consumers customizing it can just say Partial<IOverlayStyles> if they're merging with the existing one

Reply:
In this case, having them required prevents customizing the styles via the `customStyles` prop.

Reply:
Could you point me to an example of Partial styles?

Reply:
I would like for us to have a way to insure any and all classNames defined in IFooStyles gets defined in the default implementation, while allowing other implementations to use a Partial<> to only have to override a subset.
I don't have any examples (we barely do any style customization right now as it is). Where is the customStyles prop used, in getStyles? Does this work?
getStyles: IStylesFunction<IFooProps, Partial<IFooStyles>>